pkg/virt-controller/services/template.go

```diff
@@ -1390,6 +1424,26 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 	}
 }
 
+func (t *templateService) doesVMIRequireAutoMemoryLimits(vmi *v1.VirtualMachineInstance) bool {
+	if _, memLimitsExists := vmi.Spec.Domain.Resources.Limits[k8sv1.ResourceMemory]; memLimitsExists {
+		return false
+	}
+
+	if !t.clusterConfig.AutoMemoryLimitsEnabled() {
+		return false
+	}
```

Comment:
nit: I think this should be checked first

Reply:
:+1: